src/components/ProductCard/ProductCard.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { RotatingLines } from 'react-loader-spinner';
+
+import { CartContext } from '../../providers/CartProvider/CartProvider';
+
+import { FavContext } from '../../providers/FavProvider/FavProvider';
+import { AddToCartButton } from '../AddToCartButton/AddToCartButton';
+import { AddToFavButton } from '../AddToFavButton/AddToFavButton';
+
+import { getProduct } from '../../api/products';
+import { getDiscount } from '../../helpers/getDiscount';
+
+import { Product } from '../../types/Product';
+import { ProductDetails } from '../../types/ProductDetails';
+import { ProductType } from '../../types/ProductType';
+
+import './ProductCard.scss';
+
+type Props = {
+  product: Product;
+};
+
+export const ProductCard: React.FC<Props> = React.memo(({ product }) => {
+  const {
+    name,
+    price,
+    discount,
+    screen,
+    capacity,
+    ram,
+    id,
+    imageUrl,
+    type,
+  } = product;
+
+  const { productsInCart, setProductsInCart } = useContext(CartContext);
+  const { favoriteProducts, setFavoriteProducts } = useContext(FavContext);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [productWithDetails, setProductWithDetails]
+    = useState<ProductDetails>();
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    const loadProduct = async () => {
+      try {
+        const productFromServer = await getProduct(product.id);
+
+        setProductWithDetails(productFromServer);
+      } catch {
+        throw new Error('Loading Error');
```

Comment:
do not throw error in `catch` block it'll cause uncaught error in your script. Do something instead. for instance, show some error message to user.